aci/resource_aci_cloudtemplateregiondetail.go

```diff
@@ -0,0 +1,183 @@
+	} else {
+		d.Set("parent_dn", GetParentDn(cloudTemplateRegionDetail.DistinguishedName, fmt.Sprintf("/"+models.RnCloudtemplateRegionDetail)))
+	}
+	d.Set("annotation", cloudTemplateRegionDetailMap["annotation"])
+	d.Set("hub_networking", toggleOptions(cloudTemplateRegionDetailMap["hubNetworkingEnabled"]))
+	return d, nil
+}
+
+func resourceAciCloudTemplateRegionImport(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[DEBUG] %s: Beginning Import", d.Id())
+	aciClient := m.(*client.Client)
+	dn := d.Id()
+	cloudTemplateRegionDetail, err := getRemoteCloudTemplateRegion(aciClient, dn)
+	if err != nil {
+		return nil, err
+	}
+	schemaFilled, err := setCloudTemplateRegionAttributes(cloudTemplateRegionDetail, d)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Printf("[DEBUG] %s: Import finished successfully", d.Id())
+	return []*schema.ResourceData{schemaFilled}, nil
+}
+
+func resourceAciCloudTemplateRegionCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] CloudTemplateRegion: Beginning Creation")
+	aciClient := m.(*client.Client)
+	CloudProviderandRegionNamesDn := d.Get("parent_dn").(string)
+
+	cloudTemplateRegionDetailAttr := models.CloudTemplateRegionAttributes{}
+
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		cloudTemplateRegionDetailAttr.Annotation = Annotation.(string)
+	} else {
+		cloudTemplateRegionDetailAttr.Annotation = "{}"
+	}
+
+	if HubNetworkingEnabled, ok := d.GetOk("hub_networking"); ok {
+		cloudTemplateRegionDetailAttr.HubNetworkingEnabled = toggleOptions(HubNetworkingEnabled.(string))
+	}
+	cloudTemplateRegionDetail := models.NewCloudTemplateRegion(fmt.Sprintf(models.RnCloudtemplateRegionDetail), CloudProviderandRegionNamesDn, cloudTemplateRegionDetailAttr)
+
+	err := aciClient.Save(cloudTemplateRegionDetail)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(cloudTemplateRegionDetail.DistinguishedName)
+	log.Printf("[DEBUG] %s: Creation finished successfully", d.Id())
+	return resourceAciCloudTemplateRegionRead(ctx, d, m)
+}
+func resourceAciCloudTemplateRegionUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] CloudTemplateRegion: Beginning Update")
+	aciClient := m.(*client.Client)
+	CloudProviderandRegionNamesDn := d.Get("parent_dn").(string)
+
+	cloudTemplateRegionDetailAttr := models.CloudTemplateRegionAttributes{}
+
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		cloudTemplateRegionDetailAttr.Annotation = Annotation.(string)
+	} else {
+		cloudTemplateRegionDetailAttr.Annotation = "{}"
+	}
+
+	if HubNetworkingEnabled, ok := d.GetOk("hub_networking"); ok {
+		cloudTemplateRegionDetailAttr.HubNetworkingEnabled = toggleOptions(HubNetworkingEnabled.(string))
+	}
+	cloudTemplateRegionDetail := models.NewCloudTemplateRegion(fmt.Sprintf(models.RnCloudtemplateRegionDetail), CloudProviderandRegionNamesDn, cloudTemplateRegionDetailAttr)
```

Comment:
This seems so to be repetitive code, should we create function for this? And use in create/update?

Reply:
ok